### PR TITLE
[Merged by Bors] - chore(Topology/Constructions): rename most type variables

### PR DESCRIPTION
**Pull request by grunweg**
Now we use letters X, Y, Z and W for topological spaces, not Greek letters.

Also rename associated variables (a : X) to (x : X), etc., and hypotheses.
I didn't replace all letters; some lemmas need a large number of different letters. Volunteers for the last instances welcome.

---

Best reviewed commit by commit.

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Topology/Constructions.lean [hunk 1]
```diff
@@ -39,39 +39,39 @@ open Topology TopologicalSpace Set Filter Function Classical
...
 
-instance instTopologicalSpaceSum [t₁ : TopologicalSpace α] [t₂ : TopologicalSpace β] :
-    TopologicalSpace (α ⊕ β) :=
+instance instTopologicalSpaceSum [t₁ : TopologicalSpace X] [t₂ : TopologicalSpace Y] :
+    TopologicalSpace (X ⊕ Y) :=
   coinduced Sum.inl t₁ ⊔ coinduced Sum.inr t₂
 
-instance instTopologicalSpaceSigma {β : α → Type v} [t₂ : ∀ a, TopologicalSpace (β a)] :
-    TopologicalSpace (Sigma β) :=
+instance instTopologicalSpaceSigma {X : X → Type v} [t₂ : ∀ a, TopologicalSpace (X a)] :
```
Comment (urkud) [inline Mathlib/Topology/Constructions.lean:65]: It should be `X : α → Type*`, not `X : X → Type*`.

File: Mathlib/Topology/Constructions.lean [hunk 2]
```diff
@@ -39,39 +39,39 @@ open Topology TopologicalSpace Set Filter Function Classical
...
+    TopologicalSpace (Sigma X) :=
   ⨆ a, coinduced (Sigma.mk a) (t₂ a)
 
-instance Pi.topologicalSpace {β : α → Type v} [t₂ : (a : α) → TopologicalSpace (β a)] :
-    TopologicalSpace ((a : α) → β a) :=
+instance Pi.topologicalSpace {Y : X → Type v} [t₂ : (a : X) → TopologicalSpace (Y a)] :
```
Comment (urkud) [inline Mathlib/Topology/Constructions.lean:69]: Same here: the index type has no topology, so it shouldn't be named `X`.

File: Mathlib/Topology/Constructions.lean [hunk 3]
```diff
@@ -152,125 +152,125 @@ The topology on this type synonym is inherited without change.
...
 
-instance {p : α → Prop} [TopologicalSpace α] [DiscreteTopology α] : DiscreteTopology (Subtype p) :=
+instance {p : X → Prop} [TopologicalSpace X] [DiscreteTopology X] : DiscreteTopology (Subtype p) :=
   ⟨bot_unique fun s _ => ⟨(↑) '' s, isOpen_discrete _, preimage_image_eq _ Subtype.val_injective⟩⟩
 
-instance Sum.discreteTopology [TopologicalSpace α] [TopologicalSpace β] [hα : DiscreteTopology α]
-    [hβ : DiscreteTopology β] : DiscreteTopology (Sum α β) :=
-  ⟨sup_eq_bot_iff.2 <| by simp [hα.eq_bot, hβ.eq_bot]⟩
+instance Sum.discreteTopology [TopologicalSpace X] [TopologicalSpace Y] [h : DiscreteTopology X]
+    [hY : DiscreteTopology Y] : DiscreteTopology (Sum X Y) :=
```
Comment (urkud) [inline Mathlib/Topology/Constructions.lean:209]:
```suggestion
    [hY : DiscreteTopology Y] : DiscreteTopology (X ⊕ Y) :=
```

File: Mathlib/Topology/Constructions.lean (hunk 1 above)
Comment (grunweg) [reply Mathlib/Topology/Constructions.lean]: Sure. I've changed to use \iota as indexing type (as many other files also do).

File: Mathlib/Topology/Constructions.lean (hunk 3 above)
Comment (grunweg) [reply Mathlib/Topology/Constructions.lean]: I also just pushed a commit doing the same replacement across this file. (Happy to pull into its own PR.)

**Comment by grunweg** [comment]
Thanks for the careful comments. I have updated the code accordingly.
awaiting-review

**Comment by urkud** [comment]
A few more comments:

- the second argument of `IsClosed.setOf_mapsTo` is not a topological space;
- `Inducing.prod_map` still uses `δ` for a topological space; I don't know what letter should it use instead;
- same for `Embedding.prod_map` and `IsOpenMap.prod`, maybe more (search for `δ`);

Otherwise LGTM. Thanks!

bors d+

**Comment by grunweg** [comment]
Thanks again for the review. I pushed some last changes (renaming misleading variables); let me know if you prefer some changes to these (happy to make them after the fact).
bors r+

**Pull request closed without merging**